lib/transport.js

```diff
@@ -0,0 +1,19 @@
+'use strict'
+
+const { Worker } = require('worker_threads')
+
+function transport (filePath, args = []) {
+  args.unshift(filePath)
+  const code = `
+    process.stdin.fd = 1
+    process.argv.push(...${JSON.stringify(args)})
```

Comment:
I don’t know where we set with this in core, but it still under debate if a thread doing this will not alter the main thread.

Reply:
so, if you're not using the `eval` option you can pass `execArgv` and it overrides.
Using `eval` clears all the args (because it's like repl mode), and setting it doesn't affect the parent.